cmd/cstor-pool-mgmt/controller/replica-controller/handler.go

```diff
@@ -0,0 +1,205 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package replicacontroller
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+
+	"github.com/golang/glog"
+	"github.com/openebs/maya/cmd/cstor-pool-mgmt/controller/common"
+	"github.com/openebs/maya/cmd/cstor-pool-mgmt/volumereplica"
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/cache"
+)
+
+// syncHandler compares the actual state with the desired, and attempts to
+// converge the two. It then updates the Status block of the CStorReplicaUpdated resource
+// with the current status of the resource.
+func (c *CStorVolumeReplicaController) syncHandler(key, operation string) error {
+	cVRGot, err := c.getVolumeReplicaResource(key, operation)
+	if err != nil {
+		return err
+	}
+	status, err := c.cVREventHandler(operation, cVRGot)
+	if status == "" {
+		return nil
+	}
+	cVRGot.Status.Phase = apis.CStorVolumeReplicaPhase(status)
+	if err != nil {
+		glog.Errorf(err.Error())
+		glog.Infof("cVR:%v, %v; Status: %v", cVRGot.Name,
+			string(cVRGot.GetUID()), cVRGot.Status.Phase)
+		_, err := c.clientset.OpenebsV1alpha1().CStorVolumeReplicas().Update(cVRGot)
+		if err != nil {
+			return err
+		}
+		return err
+	}
+	_, err = c.clientset.OpenebsV1alpha1().CStorVolumeReplicas().Update(cVRGot)
+	if err != nil {
+		return err
+	}
+	glog.Infof("cVR:%v, %v; Status: %v", cVRGot.Name,
+		string(cVRGot.GetUID()), cVRGot.Status.Phase)
+	return nil
+
+}
+
+func (c *CStorVolumeReplicaController) cVREventHandler(operation string, cVR *apis.CStorVolumeReplica) (string, error) {
+	err := volumereplica.CheckValidVolumeReplica(cVR)
+	if err != nil {
+		c.recorder.Event(cVR, corev1.EventTypeWarning, common.FailureValidate, common.MessageResourceFailValidate)
+		return string(apis.CVRStatusOffline), err
+	}
+
+	// PoolNameHandler tries to get pool name and blocks for
+	// particular number of attempts.
+	var noOfAttempts = 2
+	if !common.PoolNameHandler(cVR, noOfAttempts) {
+		return string(apis.CVRStatusOffline), fmt.Errorf("Pool not present")
+	}
+
+	fullVolName := "cstor-" + cVR.Labels["cstorpool.openebs.io/uid"] + "/" + cVR.Labels["cstorvolume.openebs.io/uid"]
+	glog.Infof("fullVolName: %v", fullVolName)
+
+	switch operation {
+	case "add":
+		glog.Infof("Processing cvr added event: %v, %v", cVR.ObjectMeta.Name, string(cVR.GetUID()))
+
+		// To check if volume is already imported with pool.
+		importedFlag := common.CheckForInitialImportedPoolVol(common.InitialImportedPoolVol, fullVolName)
+		if importedFlag && !IsInitStatus(cVR) {
+			glog.Infof("CStorVolumeReplica %v is already imported", string(cVR.ObjectMeta.UID))
+			c.recorder.Event(cVR, corev1.EventTypeNormal, common.SuccessImported, common.MessageResourceImported)
+			return string(apis.CVRStatusOnline), nil
+		}
+
```

Comment:
Add a check to handle the add event for a duplicate resource.

Reply:
will change